application/event_processor/dos.py

```diff
@@ -67,80 +68,119 @@ def __repr__(self) -> str:
+            self._specififed_opening_times = get_specified_opening_times_from_db(self.id)
+        return self._specififed_opening_times
 
 
-def add_address_to_change_request_if_not_equal(
-    changes: dict, change_key: str, dos_address: str, nhs_uk_entity: NHSEntity
-) -> dict:
-    """Adds the address to the change request if the address is not equal
+
+def get_matching_dos_services(odscode: str) -> List[DoSService]:
+    """Retrieves DoS Services from DoS database
 
     Args:
-        changes (dict): Change Request changes
-        change_key (str): Key to add to the change request
-        dos_address (str): Address from the DoS database for comparision
-        nhs_uk_entity (NHSEntity): NHS UK Entity for comparision
+        odscode (str): ODScode to match on
 
     Returns:
-        dict: Change Request changes
+        list[DoSService]: List of DoSService objects with matching first 5
+        digits of odscode, taken from DoS database
     """
-    nhs_uk_address_lines = [
-        nhs_uk_entity.Address1,
-        nhs_uk_entity.Address2,
-        nhs_uk_entity.Address3,
-        nhs_uk_entity.City,
-        nhs_uk_entity.County,
-    ]
-    nhs_uk_address = [address for address in nhs_uk_address_lines if address is not None and address.strip() != ""]
-    nhs_uk_address_string = "$".join(nhs_uk_address)
-    if dos_address != nhs_uk_address_string:
-        logger.debug(f"Address is not equal, {dos_address=} != {nhs_uk_address_string=}")
-        changes[change_key] = nhs_uk_address
-    return changes
 
+    logger.info(f"Searching for DoS services with ODSCode that matches first "
+                f"5 digits of '{odscode}'")
 
-def get_matching_dos_services(odscode: str) -> List[DoSService]:
-    """Retrieves DoS Services from DoS database
+    sql_command = f"SELECT {', '.join(DoSService.db_columns)} FROM services WHERE odscode LIKE '{odscode[0:5]}%'"
+    logger.info(f"Created SQL command to run: {sql_command}")
+    c = query_dos_db(sql_command)
 
-    Args:
-        odscode (str): ODScode to match on
+    # Create list of DoSService objects from returned rows
+    services = [DoSService(row) for row in c.fetchall()]
+    c.close()
+    return services
 
+
+def get_specified_opening_times_from_db(service_id: int) -> List[SpecifiedOpeningTime]:
+    """Retrieves specified opening times from  DoS database
+    Args:
+        serviceid (int): serviceid to match on
     Returns:
-        list[DoSService]: List of DoSService objects with matching first 5 digits of odscode, taken from DoS database
+        List[SpecifiedOpeningTime]: List of Specified Opening times with 
+        matching serviceid
     """
 
-    logger.info(f"Searching for DoS services with ODSCode that matches first 5 digits of '{odscode}'")
+    logger.info(f"Searching for specified opening times with serviceid that "
+                f"matches '{service_id}'")
+
+    sql_command = ("SELECT ssod.serviceid, ssod.date, ssot.starttime, "
```

Comment:
in each query, information is coming from  two different tables and it might be tricky  to combine these two queries but I will test in the workbench first before I change it

Reply:
Each db query deals with separate datasets. While theoretically we could have one giant Cursor returned from a very complex SQL command for everything we need per service. It would add an amount of complexity to the code to extract it properly. The connection to the DB is currently kept for all queries so the speed isn't impacted too much. Unless there was a requirement to reduce db hits or reduce speed drastically then adding in the complexity doesn't seem worthwhile.